### PR TITLE
Add average precision

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Still to clean-up, document and test (corner cases)

This still uses the confidence steps + non-greedy method. 
I have to look if going one by one for all predictions and using greedy matching would be better.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@aboucaud Can you take a look at this?

File: workflow/scores/curves.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    """
+    ps = np.asarray(ps)
+    rs = np.asarray(rs)
+
+    p_at_r = []
+
+    for r in np.arange(0, 1.1, 0.1):
+        p = np.array(ps)[np.array(rs) >= r]
+        if p.size:
```
Comment (aboucaud) [inline workflow/scores/curves.py:138]: If I want to be peeky (there's nothing more to say, so..) I would rather use `np.linspace(0, 1, 11)` 😃

**Review by aboucaud — COMMENTED** [review]
LGTM
